Formula/ccache.rb

```diff
@@ -15,9 +14,19 @@ class Ccache < Formula
     sha256 cellar: :any_skip_relocation, x86_64_linux:  "2bf3bb826da52e61604f2ab670ecff55b7834306e9b6563b0f1b8d7eacb30e89"
   end
 
+  depends_on "asciidoctor" => :build
   depends_on "cmake" => :build
+  depends_on "pkg-config" => :build
+
+  depends_on "hiredis"
   depends_on "zstd"
 
+  on_linux do
+    depends_on "gcc" => :build
```

Comment:
Some of the libstdc++ linkage will need `gcc` as runtime dependency. It looks like CI doesn't properly test this right now. `gcc-5` only has GLIBCXX_3.4.21, but the binary needs GLIBCXX_3.4.29:
```console
❯ objdump -p bin/ccache | rg GLIBCXX
    0x0297f870 0x00 25 GLIBCXX_3.4.20
    0x0297f867 0x00 24 GLIBCXX_3.4.17
    0x0297f869 0x00 23 GLIBCXX_3.4.19
    0x0297f879 0x00 19 GLIBCXX_3.4.29
    0x0297f868 0x00 17 GLIBCXX_3.4.18
    0x02297f89 0x00 16 GLIBCXX_3.4.9
    0x0297f876 0x00 15 GLIBCXX_3.4.26
    0x0297f861 0x00 14 GLIBCXX_3.4.11
    0x0297f864 0x00 12 GLIBCXX_3.4.14
    0x0297f872 0x00 07 GLIBCXX_3.4.22
    0x0297f871 0x00 06 GLIBCXX_3.4.21
    0x0297f865 0x00 05 GLIBCXX_3.4.15
    0x08922974 0x00 02 GLIBCXX_3.4
```

Installing this on local Docker and running `brew test` results in:
```
==> Testing ccache
==> which gcc
==> /home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache -s
Last 15 lines from /root/.cache/Homebrew/Logs/ccache/test.01.ccache:
2021-08-21 08:42:44 +0000

/home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache
-s

/home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.29' not found (required by /home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache)
/home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.26' not found (required by /home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache)
/home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache: /usr/lib/x86_64-linux-gnu/libstdc++.so.6: version `GLIBCXX_3.4.22' not found (required by /home/linuxbrew/.linuxbrew/Cellar/ccache/4.4/bin/ccache)
Error: ccache: failed
```

Reply:
I made gcc not only a build but also runtime dependency as suggested by @carlocab.